master/internal/rm/resource_manager_iface.go

```diff
@@ -36,7 +36,7 @@ type ResourceManager interface {
 	GetDefaultAuxResourcePool(sproto.GetDefaultAuxResourcePoolRequest) (sproto.GetDefaultAuxResourcePoolResponse, error)
 	ValidateResourcePool(name string) error
 	ResolveResourcePool(name string, workspace, slots int) (string, error)
-	ValidateResourcePoolAvailability(name string, slots int) ([]command.LaunchWarning, error)
+	ValidateResourcePoolAvailability(v *sproto.ValidateResourcePoolAvailabilityRequest) ([]command.LaunchWarning, error)
```

Comment:
I think we will also need to update ```test_log_policy_exclude_node``` for the agent case since now jobs will fail instead of stay in queued.

We probably want to keep the test the same for Kubernetes and create a new test for agents